### PR TITLE
Add unit test for searching by participant_status_id & switch to generic function

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
This creates a generic function for handling where AND the qill for fields with adequate metadata & adds tests & switches one field to it. It also contains 2 pieces of cleanup from https://github.com/civicrm/civicrm-core/pull/15791 although I have not attempted full support of url mode in this PR

Before
----------------------------------------
No single generic function exists for meta-data-ready fields

After
----------------------------------------
Fn is used & tested

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton looks like we need a rebase

**Comment by seamuslee001** [comment]
test fail unrelated

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -7201,19 +7201,44 @@ protected function getSelectedGroupStatuses($grouping) {
...
+   * @param string|array|int $value
+   * @param string $op
+   * @param string|int $grouping
+   *
+   * @throws \CRM_Core_Exception
+   */
+  public function handleWhereFromMetadata($fieldSpec, $name, $value, $op, $grouping = 0) {
+    $this->_where[$grouping][] = CRM_Contact_BAO_Query::buildClause($fieldSpec['where'], $op, $value, CRM_Utils_Type::typeToString($fieldSpec['type']));
+    $this->_qill[$grouping][] = $this->getQillForField($name, $value, $op, $fieldSpec);
+    if (!isset($query->_tables[$fieldSpec['table_name']])) {
```
Comment (jitendrapurohit) [inline CRM/Contact/BAO/Query.php]: seems `$query` is undefined here? Should it be `$this`?

Comment (eileenmcnaughton) [reply CRM/Contact/BAO/Query.php]: opps yes!

Comment (eileenmcnaughton) [reply CRM/Contact/BAO/Query.php]: https://github.com/civicrm/civicrm-core/pull/15865

**Pull request merged**